lib/src/discovery.dart

```diff
@@ -53,10 +53,10 @@ Future<PackageConfig /*?*/ > findPackageConfig(
 }
 
 /// Similar to [findPackageConfig] but based on a URI.
-Future<PackageConfig /*?*/ > findPackageConfigUri(
+Future<PackageConfig?> findPackageConfigUri(
     Uri location,
-    Future<Uint8List /*?*/ > loader(Uri uri) /*?*/,
-    void onError(Object error) /*?*/,
+    Future<Uint8List?> loader(Uri uri)?,
```

Comment:
It is pretty weird to have a nullable required positional argument like this... should we consider changing the api?

It seems like the only required argument should be `location` really and the others should be optional named args.

Reply:
It's internal API. 
I usually don't bother with optional parameters for internal API, it's not there to be ergonomic, just to be clear and precise.